test/e2e/lvmcluster.go

```diff
@@ -84,3 +93,106 @@ func deleteLVMCluster(ctx context.Context) error {
 
 	return err
 }
+
+func setupPodAndPVC() (*corev1.PersistentVolumeClaim, *corev1.Pod) {
+	filePvcYaml := fmt.Sprintf(pvcYAMLTemplate, "lvmfilepvc", testNamespace, "Filesystem", storageClassName)
+	pvc, err := getPVC(filePvcYaml)
+	Expect(err).To(BeNil())
+
+	err = crClient.Create(context.Background(), pvc)
+	Expect(err).To(BeNil())
+
+	podVolumeMountYaml := fmt.Sprintf(podVolumeFSYAMLTemplate, "lvmfilepod", testNamespace, "lvmfilepvc")
+	pod, err := getPod(podVolumeMountYaml)
+	Expect(err).To(BeNil())
+
+	err = crClient.Create(context.Background(), pod)
+	Expect(err).To(BeNil())
+
+	return pvc, pod
+}
+
+func cleanupPVCAndPod(pvc *corev1.PersistentVolumeClaim, pod *corev1.Pod) {
+	err := crClient.Delete(context.Background(), pod)
+	Expect(err).To(BeNil())
+	fmt.Printf("Pod %s is deleted\n", pod.Name)
+
+	err = crClient.Delete(context.Background(), pvc)
+	Expect(err).To(BeNil())
+	fmt.Printf("PVC %s is deleted\n", pvc.Name)
+}
+
+func lvmClusterTest() {
+	Describe("Filesystem Type", Serial, func() {
+
+		var clusterConfig *v1alpha1.LVMCluster
+		ctx := context.Background()
+
+		AfterEach(func() {
+			// Delete the cluster
+			lvmClusterCleanup(clusterConfig, ctx)
+		})
+
+		It("should default to xfs", func() {
+			clusterConfig = generateLVMCluster() // Do not specify a fstype
+
+			By("Setting up the cluster with the default fstype")
+			lvmClusterSetup(clusterConfig, ctx)
+
+			// Make sure the storage class was configured properly
+			sc := storagev1.StorageClass{}
+
+			Eventually(func() bool {
+				err := crClient.Get(ctx, types.NamespacedName{Name: storageClassName, Namespace: installNamespace}, &sc)
+				if err != nil {
+					debug("Error getting StorageClass %s: %s\n", storageClassName, err.Error())
+				}
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+
+			Expect(sc.Parameters["csi.storage.k8s.io/fstype"]).To(Equal(string(v1alpha1.FilesystemTypeXFS)))
+		})
+
+		It("should be xfs if specified", func() {
+			clusterConfig = generateLVMCluster()
+			clusterConfig.Spec.Storage.DeviceClasses[0].FilesystemType = v1alpha1.FilesystemTypeXFS
```

Comment:
Question for my own edification...why is `DeviceClasses[0]` selected here but not in the default case? Is it just the case where if the user sets it specifically they must have set `spec.Storage.DeviceClasses[0]`?

Reply:
Yeah they could in theory set multiple device classes. Each device class will have a default value of XFS if one isn't specified. So this is a test case where it would be specified as XFS rather than using the default value